### ISSUE TITLE
Failed to parse target a-b:1234

### ISSUE THREAD

**Issue by merlinnot** [grpc/grpc-node#1044]
### Problem description

Not all valid hostnames can be used with `@grpc/grpc-js` v0.6.0, e.g. `a-b`, `a.b` are not considered valid given this RegExp: https://github.com/grpc/grpc-node/blob/e055468c299a1a22cc36b7760175b1964a3067d7/packages/grpc-js/src/resolver-dns.ts#L57

### Reproduction steps

Try to initialize the client with a hostnames like this:
- `firestore-emulator:1234`
- `http://pubsub-emulator:4321`

### Environment
 - OS name, version and architecture: Darwin Kernel Version 19.0.0: Sat Aug 31 18:49:12 PDT 2019; root:xnu-6153.11.15~8/RELEASE_X86_64 x86_64
 - Node version: 10.16.3
 - Node installation method: nvm
 - If applicable, compiler version: Apple clang version 11.0.0 (clang-1100.0.31.5)
 - Package name and version: @grpc/grpc-js@0.6.0

### Additional context

```
Step #3 - "test":     Failed to parse target http://firestore-emulator:4500
Step #3 - "test": 
Step #3 - "test":       38 |     const sslCredentials = credentials.createInsecure();
Step #3 - "test":       39 | 
Step #3 - "test":     > 40 |     const client = new FirestoreEmulator(address, sslCredentials);
Step #3 - "test":          |                    ^
Step #3 - "test":       41 | 
Step #3 - "test":       42 |     return client;
Step #3 - "test":       43 |   },
Step #3 - "test": 
Step #3 - "test":       at Function.getDefaultAuthority (node_modules/@grpc/grpc-js/build/src/resolver-dns.js:242:15)
Step #3 - "test":       at Object.getDefaultAuthority (node_modules/@grpc/grpc-js/build/src/resolver.js:72:32)
Step #3 - "test":       at new ChannelImplementation (node_modules/@grpc/grpc-js/build/src/channel.js:85:48)
Step #3 - "test":       at new Client (node_modules/@grpc/grpc-js/build/src/client.js:37:36)
Step #3 - "test":       at new ServiceClientImpl (node_modules/@grpc/grpc-js/build/src/make-client.js:49:5)
Step #3 - "test":       at Object.<anonymous>.exports.myFunctionName (some/path/index.ts:40:20)
```

**Comment by murgatroid99** [comment]
Thank you for reporting this quickly. I'll have a fix out soon.

**Comment by murgatroid99** [comment]
I have published version 0.6.1 with the fix for this.
